OnboardIos/OnboardIos/Models/Users.m

```diff
@@ -31,74 +40,124 @@ + (NSArray *)list:(NSNumber *)page{
+    [u9 setValue:@9 forKey:@"id"];
     [u9 setValue:@"Felipe" forKey:@"first_name"];
     [u9 setValue:@"souza" forKey:@"last_name"];
     [u9 setValue:@"2https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" forKey:@"avatar"];
     [users addObject:u9];
     
     NSMutableDictionary *u10 = [NSMutableDictionary new];
-    [u10 setValue:@2 forKey:@"id"];
+    [u10 setValue:@10 forKey:@"id"];
     [u10 setValue:@"joao" forKey:@"first_name"];
     [u10 setValue:@"das neves" forKey:@"last_name"];
     [u10 setValue:@"2https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" forKey:@"avatar"];
     [users addObject:u10];
     
+    NSMutableDictionary *u11 = [NSMutableDictionary new];
+    [u11 setValue:@10 forKey:@"id"];
+    [u11 setValue:@"rafekl" forKey:@"first_name"];
+    [u11 setValue:@"matos" forKey:@"last_name"];
+    [u11 setValue:@"2https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" forKey:@"avatar"];
+    [users addObject:u11];
+    NSMutableDictionary *u12 = [NSMutableDictionary new];
+    [u12 setValue:@10 forKey:@"id"];
+    [u12 setValue:@"guiilernmes" forKey:@"first_name"];
+    [u12 setValue:@"afonso" forKey:@"last_name"];
+    [u12 setValue:@"2https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" forKey:@"avatar"];
+    [users addObject:u12];
+    NSMutableDictionary *u13 = [NSMutableDictionary new];
+    [u13 setValue:@10 forKey:@"id"];
+    [u13 setValue:@"mikey" forKey:@"first_name"];
+    [u13 setValue:@"pateat" forKey:@"last_name"];
+    [u13 setValue:@"2https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg" forKey:@"avatar"];
+    [users addObject:u13];
     return users;
 }
 
 - (void)incrementViewCount:(NSNumber *)identifier {
     DLog();
+    
+    BOOL create = YES;
+    for (int a = 0; a < userViewCount.count; a++) {
+        NSDictionary *dict = [userViewCount objectAtIndex:a];
+        if (dict[@"id"]==identifier) {
+            int increment = [dict[@"count"] intValue] + 1;
+            NSNumber *numb = [NSNumber numberWithInt:increment];
+            [dict setValue:numb forKey:@"count"];
+            create = NO;
+        }
+    }
+    
+    if (create) { //criar se nao existir
+        NSMutableDictionary *dict = [NSMutableDictionary new];
+        [dict setValue:identifier forKey:@"id"];
+        [dict setValue:@1 forKey:@"count"];
+        [userViewCount addObject:dict];
+    }
 }
 
 - (void)resetViewCount:(NSNumber *)identifier {
     DLog();
+    
+    for (int a = 0; a < userViewCount.count; a++) {
+        NSDictionary *dict = [userViewCount objectAtIndex:a];
+        if (dict[@"id"]==identifier) {
+            [dict setValue:@0 forKey:@"count"];
+        }
+    }
 }
 
 - (NSNumber *)getViewCount:(NSNumber *)identifier {
     DLog();
+    NSLog(@"%@", @"haahhashjdsjdjhdk");
+    for (int a = 0; a < userViewCount.count; a++) {
+        NSDictionary *dict = [userViewCount objectAtIndex:a];
+        if (dict[@"id"]==identifier) {
```

Comment:
Just as a warning that this `==` comparison works here only because you not comparing values, but pointers (`userViewCount` is an array of the same `NSDictionary`s memory objects where you get the `@"id"` key from)

Keep in mind that if you need to compare `string` values you would have to use explicit methods to do it [1] otherwise you might have an unexpected behavior


[1] http://stackoverflow.com/a/7266238/429521